jep/7/README.adoc

```diff
@@ -67,22 +69,68 @@ endif::[]
+Plugins depending on the either the Ruby runtime or the Python SDK will be also marked for adoption so that users can step up and take over maintenance (by switching to the regular Java runtime) without a 2 weeks waiting delay.
 
-Plugins with a mandatory dependency on `ruby-runtime` will be added to the same file, as they will not be installable without `ruby-runtime` being available.
+Plugins with a mandatory dependency on the runtimes will be added to the same file, as they will not be installable without the respective runtimes being available.
+The multiple Git repositories holding the `ruby-runtime` and Python SDK source code will be archived on GitHub.
 
-The multiple Git repositories holding the `ruby-runtime` source code will be archived on GitHub.
+Support for Ruby Runtime in the Jenkins core will be removed in future Jenkins versions.
+This includes but not limited to:
 
-Workarounds in Jenkins core implemented specifically to enable `ruby-runtime` will be reviewed and considered for reversal.
-No future core development will consider the impact on `ruby-runtime` and plugins based on it, and bug reports to core and related components about their impact on `ruby-runtime` and plugins based on it will be closed as *Won't Fix*.
+* link:https://github.com/stapler/stapler/tree/master/jruby[Stapler JRuby library] bundled in the Jenkins core
+* Permit lists for link:/jep/200[JEP-200: Switch Remoting/XStream denylist to a permitlist]
+* XStream converters for JRuby components
 
+No future core development will consider the impact on Ruby/Python runtimes and plugins based on them.
+Bug reports to core and related components about their impact on `ruby-runtime` and plugins based on it will be closed as *Won't Fix*.
+
+=== Rollout plan
+
+At least 1 month before removal::
+
+* Announcement blogpost on jenkins.io with the user-focused overview of the change and the delivery plan
+* Plugins are marked for adoption
+* Plugins will be deprecated in the Update site immediately after the announcement.
+  The deprecation notice will reference the announcement blogpost as justification.
+* Plugin users will have 1 month to bring up their concerns, if any.
+  They will have an option to step up and contribute to the plugins so that they are migrated to the standard Jenkins plugin runtime.
+
+Removal date::
+
+* Plugins are removed from official Jenkins update centers.
+  Users of both weekly and LTS release baselines will no longer be able to install or update the plugins.
+* Maybe: Additional announcement is published on jenkins.io to advice deletion of the plugins
+* If there are any unbpublished security issues related to the depublished plugins,
+  a security advisory is released for them shortly after depublishing.
+* Ruby and Python plugin runtime codebases are archived on GitHub.
+
+After the removal::
+
+A pull request is submitted to the Jenkins, with a cleanup of the code and functionality slated for deletion.
```

Comment:
As above, this seems like a lot compare to "we stop caring and may clean this up at some point".

Reply:
```suggestion
A pull request may be submitted to the Jenkins core at some point, with a cleanup of the code and functionality slated for deletion.
```